valhalla/baldr/pathlocation.h

```diff
@@ -161,7 +161,7 @@ struct PathLocation : public Location {
     }
 
     auto* filtered_edges = l->mutable_filtered_edges();
-    for (const auto& e : pl.edges) {
+    for (const auto& e : pl.filtered_edges) {
```

Comment:
this was the copy paste fail that makes it so the same edge appears twice in the list that the astar algorithm gets. this happens when we decide to run the route again with relaxed constraints: https://github.com/valhalla/valhalla/blob/86b947c744484c6694ae81d019a322b578d224c7/src/thor/route_action.cc#L321-L322

you can see it says well somethign went wrong with the routing so we're going to allow all the edge candidates to find a better route (either because it failed or because we didnt like the ferry on the pedestrian route)